src/Platforms/AbstractPlatform.php

```diff
@@ -76,6 +77,8 @@ abstract class AbstractPlatform
 
     public const CREATE_FOREIGNKEYS = 2;
 
+    protected const VALID_TIME_PRECISIONS = [];
```

Comment:
I'd consider overriding constants a bad habit. We don't do this at the moment and I believe we should not start this now. This should be a method instead.

Apart from that, does it really make sense to make this an array? A simple integer that represents the maximum precision should be sufficient, shouldn't it?

Reply:
Okay, I'll amend so we don't override constants. 

I used an array because of the comment in #2873 about SQLite supporting 0 or 3 decimal digits. As this is not relevant to the column definition in SQLite, we can do away with the array and use a method returning an int as you suggest. 

Since some databases drop zero as a precision specifier, we need to use DATETIME instead of DATETIME(0) on certain platforms so I've also added an `ignoreZeroTimePrecision()` method (which the array could have dealt with).  

Covered in a new patch.